autorest/azure/async_test.go

```diff
@@ -1096,6 +1097,42 @@ func TestFuture_Marshalling(t *testing.T) {
 	}
 }
 
+func TestFuture_WaitForCompletion(t *testing.T) {
+	r1 := newAsynchronousResponse()
+	r1.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+	r2 := newProvisioningStatusResponse("busy")
+	r2.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+	r3 := newProvisioningStatusResponse(operationSucceeded)
+	r3.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+
+	sender := mocks.NewSender()
+	sender.AppendResponse(r1)
+	sender.AppendAndRepeatResponse(r2, 2)
+	sender.AppendResponse(r3)
+
+	future := NewFuture(mocks.NewRequest())
+
+	client := autorest.Client{
+		PollingDelay:    autorest.DefaultPollingDelay,
+		PollingDuration: autorest.DefaultPollingDuration,
+		RetryAttempts:   autorest.DefaultRetryAttempts,
+		RetryDuration:   30 * time.Second,
```

Comment:
>RetryDuration:   30 * time.Second, [](start = 2, length = 34)

How much time does this test take?

Reply:
It is fast, but now you got me thinking it's not comprehensive enough, i.e. it doesn't test the retry-due-to-failure path.  I will update the test.

Reply:
Fixed.